src/mkfs.fat.c

```diff
@@ -1569,6 +1572,15 @@ int main(int argc, char **argv)
 	    }
 	    break;
 
+    case OPT_OFFSET:
+        errno = 0;
+        part_offset = (off_t) strtoll(optarg, &tmp, 0);
+        if (part_offset < 0 || !*optarg || *tmp || errno) {
+            printf("Offset must be a positive number in bytes: %s\n", optarg);
```

Comment:
Not "positive", but rather "non-negative" as zero offset is valid and the mostly used.

Reply:
great! but you have put fix into wrong commit.